Show splash screen (or blank if theme has none) on shutdown/reboot

bcat opened the pull request:
As discussed w/ @xonglebongle and @VagueParade on Discord. The new shutdown flow takes a couple seconds to complete (a little longer if some process has a lot of state to flush to disk), so having immediate visual feedback when the user decides to shut down helps a lot. Also, splash screens are pretty. 😄

This PR updates the shutdown flow to run `muxsplash` (not a thing yet, but I hear it will be soon 😃) with `shutdown.bmp` or `reboot.bmp` if it exists. And if the theme doesn't provide a splash image, we blank the display instead, which still makes the shutdown process feel more responsive (even if it's actually taking just as long as before).

Couple notes for posterity: doing this in `halt.sh` so we get the same behavior regardless of how shutdown is triggered (e.g., for the "emergency reboot" hotkey I'm planning to add soon). Also, not using existing binaries `fbv` (too slow) or `fbsplash` (only PPM support) since they won't handle rotated displays correctly.

I added some helper functions in `func.sh` to cut down on the `dispdbg` boilerplate, hope that's cool. I also updated the existing `dispdbg` uses in the device scripts to call the helpers instead. Adds a bunch of files to the PR unfortunately; however, that's in its [own commit](https://github.com/MustardOS/internal/pull/150/commits/959ba9c4b8930147eca63133a8d4caa927166170), so if you want it separately or not at all, you can just discard that one commit when merging.

xonglebongle commented:
All done! See, https://github.com/MustardOS/frontend/commit/b6c94529b3dddb156ca9e635f11bbb713f446906

I have also attached the binary for easy use, just pop it in `/opt/muos/extra/` and give it the full path of the PNG files of `shutdown.png` or `reboot.png` which will live in the main root of `<theme>/image`.

xonglebongle commented:
I should also mention that `muxsplash` will display a blank screen if an unsupported or missing file is found.

bcat commented on script/system/halt.sh:
```diff
@@ -5,23 +5,38 @@
 case "$1" in
-	halt|poweroff|reboot) ;;
+	halt|poweroff)
+		SPLASH_IMG="$GC_STO_THEME/MUOS/theme/active/image/shutdown.bmp"
```
Done.

bcat commented:
Thanks! `muxsplash` works great (much faster than `fbv`).

> I should also mention that `muxsplash` will display a blank screen if an unsupported or missing file is found.

Oh, nice! Technically, it wasn't quite the same behavior since the `DISPLAY_WRITE` I used before turned off the backlight, rather than just showing a blank image with the backlight on.

However, I think it doesn't really matter. :) Presumably most themes will eventually provide splash screens, and a couple seconds of cleared display with backlight on won't hurt anyone, so I removed the `DISPLAY_WRITE` to simplify the halt logic a bit. (There's enough subtle points for just the bare essentials.)

That does bring into question the `dispdbg` refactor in this PR, since `halt.sh` doesn't use it anymore. :) I removed those commits from this PR and will send them in a separate one for cleanliness.

bcat commented:
I also fixed up some `halt.sh` comments a bit while I'm in here (didn't really want another PR just for that...).
